File: app/src/main/java/uk/co/sundroid/util/astro/MoonPhaseEvent.java
package uk.co.sundroid.util.astro;

import java.util.Calendar;

public class MoonPhaseEvent {
	
	/** Phase. One of full, new, FQ, LQ. */
	private MoonPhase phase;
	
	/** Time the event occurs. */
	private Calendar time;
	
	public MoonPhaseEvent(MoonPhase phase, Calendar time) {
		this.phase = phase;
		this.time = time;
	}
	
	public double getPhaseDouble() {
		if (phase == MoonPhase.NEW) {
			return 0d;
		} else if (phase == MoonPhase.FIRST_QUARTER) {
			return 0.25d;
		} else if (phase == MoonPhase.FULL) {
			return 0.5d;
		} else {
			return 0.75d;
		}
	}

	public MoonPhase getPhase() {
		return phase;
	}

	public Calendar getTime() {
		return time;
	}
	
}

File: app/src/main/kotlin/uk/co/sundroid/util/astro/MoonPhaseEvent.kt
package uk.co.sundroid.util.astro

import java.util.Calendar

class MoonPhaseEvent(
        /** Phase. One of full, new, FQ, LQ.  */
        val phase: MoonPhase,
        /** Time the event occurs.  */
        val time: Calendar) {

    val phaseDouble: Double
        get() = if (phase == MoonPhase.NEW) {
            0.0
        } else if (phase == MoonPhase.FIRST_QUARTER) {
            0.25
        } else if (phase == MoonPhase.FULL) {
            0.5
        } else {
            0.75
        }

}